buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClustersPlugin.java

```diff
@@ -58,16 +59,26 @@ protected ExecOperations getExecOperations() {
         throw new UnsupportedOperationException();
     }
 
+    @Inject
+    public TestClustersPlugin(ProviderFactory providerFactory) {
+        this.providerFactory = providerFactory;
+    }
+
+    public void setRuntimeJava(Provider<File> runtimeJava) {
+        this.runtimeJavaProvider = runtimeJava;
+    }
+
     @Override
     public void apply(Project project) {
-        project.getRootProject().getPluginManager().apply(GlobalBuildInfoPlugin.class);
-        BuildParams.withInternalBuild(() -> project.getPlugins().apply(InternalDistributionDownloadPlugin.class))
-            .orElse(() -> project.getPlugins().apply(DistributionDownloadPlugin.class));
+        project.getPlugins().apply(DistributionDownloadPlugin.class);
         project.getRootProject().getPluginManager().apply(ReaperPlugin.class);
         Provider<ReaperService> reaperServiceProvider = GradleUtils.getBuildService(
             project.getGradle().getSharedServices(),
             ReaperPlugin.REAPER_SERVICE_NAME
         );
+        runtimeJavaProvider = providerFactory.provider(
```

Comment:
for public usage we just rely on a simple mechanism for declaring the JAVA RUNTIME